yt/data_objects/static_output.py

```diff
@@ -1216,6 +1216,9 @@ def setup_cosmology(self):
             w_a=w_a,
         )
 
+        if not hasattr(self, "current_time"):
+            self.current_time = self.cosmology.t_from_z(self.current_redshift)
```

Comment:
I would argue that it'd be preferable to do this in the frontend instead of the parent class. `Dataset` has already too much responsability to my taste. Plus it feels weird to have this in the common dataflow, as someone who never used a cosmological dataset. Is there something I'm missing ?

Reply:
I see what you're saying about not adding further specialism in the Dataset class, but this is inside the `setup_cosmology` method, so I don't think it goes outside of appropriate boundaries. Similar to what @jzuhone pointed out, I was anticipating that this might be useful for other cosmological frontends to avoid having duplicate subclass methods to solve the same problem.
More generally, I am quite in favor of moving some of this to generalized subclasses like `CosmologyDataset`. I think the way I did this here does not make that future work more difficult.

@neutrinoceros, for the sake of discussion, we can leave this open so you can respond if you like. If you're not convinced, I'll change it to being frontend-specific. I'm not that bothered either way.

Reply:
It being in cosmology setup seems fine, now that I continue to think about it!  :)

Reply:
Oh that's a very good point, I didn't realise it was happening in an already cosmology-specific method (or that the base Dataset class had any). Please disregard my original concern then